concepts/functions/introduction.md

```diff
@@ -0,0 +1,122 @@
+# Introduction
+
+A function allows to group code into a reusable unit.
+There are multiple ways to define functions in JavaScript.
+Here we will look at function declarations and function expressions.
+Other possibilities like [arrow functions][concept-arrow-functions] will be covered in other concepts.
+
+## Function Declaration
+
+The standard way of defining a function in JavaScript is a function declaration, also called function definition or function statement.
+
+It consists of the `function` keyword, the name of the function and a comma-separated list of parameters in round brackets.
+This is followed by the function body (the code that should be executed) wrapped in curly brackets.
+
+```javascript
+function someName(param1, param2, param3) {
+  // ...
+}
+```
+
+In JavaScript a function is invoked (called) by stating the function name followed by round brackets that contain the arguments.
+
+```javascript
+someName(arg1, arg2, arg3);
+```
+
+## Parameters
+
+When working with parameters inside the function body, be aware of possible effects to the original value that was passed to the function.
+In JavaScript, the behavior depends on the data type of the argument.
+
+- All values that have primitive data types ([full list here][mdn-primitives]) are immutable in JavaScript, so if used as arguments they are _passed by value_.
+  That means you are dealing with a copy of the original value in the function body and you can modify it without affecting the original value.
+- All other values (objects, arrays, functions) are _passed by reference_.
```

Comment:
Same comments as above, but in the introduction I would be inclined to make this section even simpler.